tensorflow/compiler/xla/mlir_hlo/tests/Dialect/mhlo/mhlo_infer_shape_type_methods.mlir

```diff
@@ -415,4 +415,62 @@ func.func @add_bounds_unranked(
   // CHECK: types0 = tensor<*xf32>
   %1 = "mhlo_test.get_return_types"(%result) : (tensor<*xf32>) -> tensor<*xindex>
   func.return %1 : tensor<*xindex>
-}
\ No newline at end of file
+}
+
+// -----
+
+// CHECK-LABEL: @reduce
+func.func @reduce(%arg0: tensor<?x4x12xf32>, %arg1: tensor<?x4x12xi32>) -> tensor<?xindex>{
+  %0 = mhlo.constant dense<-0.000000e+00> : tensor<f32>
+  %1 = mhlo.constant dense<1> : tensor<i32>
+  %2:2 = "mhlo.reduce"(%arg0, %arg1, %0, %1) ({
+  ^bb0(%arg4: tensor<f32>, %arg5: tensor<i32>, %arg6: tensor<f32>, %arg7: tensor<i32>):
+    %4 = mhlo.add %arg4, %arg6 : tensor<f32>
+    %5 = mhlo.add %arg5, %arg7 : tensor<i32>
+    "mhlo.return"(%4, %5) : (tensor<f32>, tensor<i32>) -> ()
+  }) {dimensions = dense<[1, 2]> : tensor<2xi64>}
```

Comment:
If you make the reduction dimensions here `[2]`, then the result of the reduction will have a partially dynamic shape and as a result, we should see this turn up in the return_type_components.

Reply:
add a test that results the  partially dynamic shape